docs/React.md

```diff
@@ -0,0 +1,30 @@
+# A brief guide to the React migration
+
+If you want get familiar with React, start with the [official site](reactjs.org). It has a nice [tutorial](https://reactjs.org/tutorial/tutorial.html) and [guide](https://reactjs.org/docs/hello-world.html).
+
+## What do we want to achieve?
+
+We want to have separate codebases for server and client. The client will be written in React.
+
+We'll replace the current AngularJS components one by one. Eventually, we'll drop the AngularJS and create a React project. 
+
+### Rationale
+
+- React is more attractive to developers.
+- AngularJS is being replaced by newer frameworks.
+- [MEAN is not maintained](https://github.com/Trustroots/trustroots/issues/638). 
+
+## How to wire up React components to AngularJS?
+
+Create a react component with a path `modules/**/client/components/SomeName.component.js`. It will get imported to Angular automatically. Then you can use this component within Angular like `<some-name></some-name>`.
+See an [example component](https://github.com/Trustroots/trustroots/blob/master/modules/pages/client/components/Volunteering.component.js).
+
+## Issues and solutions
+
+### Routing
+
+We're still using AngularJS for routing.
```

Comment:
Noting that we're not using AngularJS's internal router, but a bit more capable UI-Router: https://ui-router.github.io/ng1/

Reply:
```